src/main/java/uk/gov/companieshouse/search/api/service/search/impl/dissolved/DissolvedSearchRequestService.java

```diff
@@ -0,0 +1,128 @@
+			}
+
+			if (hits.getTotalHits().value == 0) {
+
+				hits = dissolvedSearchRequests
+						.getCorporateNameStartsWithResponse(orderedAlphaKey, requestId);
+			}
+
+			if (hits.getTotalHits().value > 0) {
+				LoggingUtils.getLogger().info("A result has been found", logMap);
+
+				String orderedAlphaKeyWithId = SearchRequestUtils.getOrderedAlphaKeyWithId(hits.getHits()[0]);
+				SearchHit bestMatch = hits.getHits()[0];
+				DissolvedCompany topHitCompany = mapESResponse(bestMatch);
+				topHit.setCompanyName(topHitCompany.getCompanyName());
+				topHit.setCompanyNumber(topHitCompany.getCompanyNumber());
+
+				populateSearchResults(requestId, topHit.getCompanyName(), results, topHitCompany, orderedAlphaKeyWithId);
+			}
+		} catch (IOException e) {
+			LoggingUtils.getLogger().error(DISSOLVED_SEARCH + "failed to map highest map to company object for: ", logMap);
+			throw new SearchException("error occurred reading data for highest match from " +
+					"searchHits", e);
+		}
+
+		return new DissolvedSearchResults("", topHit, results);
+	}
+
+	private DissolvedCompany mapESResponse(SearchHit hit) {
+		Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+		Map<String, Object> address = (Map<String, Object>) sourceAsMap.get("address");
+		List<PreviousCompanyName> previousCompanyNamesList = (List<PreviousCompanyName>) sourceAsMap.get("previous_company_names");
+
+		DissolvedCompany dissolvedCompany = new DissolvedCompany();
+		Address roAddress = new Address();
+
+		dissolvedCompany.setCompanyName((String) sourceAsMap.get("company_name"));
+		dissolvedCompany.setCompanyNumber((String) sourceAsMap.get("company_number"));
+		dissolvedCompany.setCompanyStatus((String) sourceAsMap.get("company_status"));
+		dissolvedCompany.setDateOfCessation((String) sourceAsMap.get("date_of_cessation"));
+		dissolvedCompany.setDateOfCreation((String) sourceAsMap.get("date_of_creation"));
+		roAddress.setLocality((String) address.get("locality"));
+		roAddress.setPostalCode((String) address.get("postal_code"));
+
+		dissolvedCompany.setAddress(roAddress);
+		dissolvedCompany.setPreviousCompanyNames(previousCompanyNamesList);
+
+		return dissolvedCompany;
+	}
+
+	private void populateSearchResults(String requestId,
+									   String topHitCompanyName,
+									   List<DissolvedCompany> results,
+									   DissolvedCompany topHitCompany,
+									   String orderedAlphaKeyWithId) throws IOException {
+		SearchHits hits;
+		hits = dissolvedSearchRequests.getAboveResultsResponse(requestId, orderedAlphaKeyWithId,
+				topHitCompanyName);
+		hits.forEach(h -> results.add(mapESResponse(h)));
+
+		Collections.reverse(results);
+
+		LoggingUtils.getLogger().info("Retrieving the top hit: " + topHitCompanyName);
+		results.add(topHitCompany);
+
+		hits = dissolvedSearchRequests.getDescendingResultsResponse(requestId, orderedAlphaKeyWithId,
+				topHitCompanyName);
+
+		hits.forEach(h -> results.add(mapESResponse(h)));
+	}
```

Comment:
The formatting of this is incorrect should use spaces instead of tabs and the method parameters shouldn't be in their present location (lines 109-112)